### ISSUE TITLE
[QST] Can't follow the Examples

### ISSUE THREAD

**Issue by bayismet** [NVIDIA-Merlin/Transformers4Rec#330]
# ❓ Questions & Help

## Details
Hello there!

I'm trying to follow the "[tutorial](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples/tutorial)" folder under the examples. But in the [third notebook](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/main/examples/tutorial/03-Session-based-recsys.ipynb), it seems I can't export my trained model. I only changed the batch sizes in my local machine and trained all three of the models (GRU, XLNET, XLNET with side information) without an error. Here is the error that I see:

`from nvtabular.inference.triton import export_pytorch_ensemble
export_pytorch_ensemble(
    model,
    workflow,
    sparse_max=sparse_features_max,
    name= "t4r_pytorch",
    model_path= "./models",
    label_columns =[],
)`

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_7895/965328198.py in <module>
----> 1 from nvtabular.inference.triton import export_pytorch_ensemble
      2 export_pytorch_ensemble(
      3     model,
      4     workflow,
      5     sparse_max=sparse_features_max,

~/anaconda3/envs/rapids-21.10/lib/python3.8/site-packages/nvtabular/inference/triton/__init__.py in <module>
     30 from tritonclient.utils import np_to_triton_dtype  # noqa
     31 
---> 32 import nvtabular.inference.triton.model_config_pb2 as model_config  # noqa
     33 from nvtabular.columns import Schema  # noqa
     34 from nvtabular.dispatch import _is_list_dtype, _is_string_dtype, _make_df  # noqa

ModuleNotFoundError: No module named 'nvtabular.inference.triton.model_config_pb2'

Thanks!

**Comment by rnyak** [comment]
@bayismet Thanks for reporting the issue. Can you give a bit details about your env? Are you using docker images or running notebooks in a conda env? what's your cuda and driver version? can you copy here the print outs of `nvidia-smi` and `nvcc --version` commands?

in the mean time can you try 

```
export PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION='python'
```

let us know if that'd work?

**Comment by bayismet** [comment]
Hello there!

I executed the command `export PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION='python'`, then open VSCode and run my notebook, but nothing is different (I'm choosing the rapids-21.10 conda enviroment). 

My "nvidia-smi" output is here (it's RTX 2060 xs):

(rapids-21.10) ibayramoglu@ibayramoglu-MS-7C02:~$ nvidia-smi
Wed Nov 17 13:03:07 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.57.02    Driver Version: 470.57.02    CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:26:00.0  On |                  N/A |
| 24%   29C    P8     9W / 160W |    196MiB /  5928MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1132      G   /usr/lib/xorg/Xorg                 23MiB |
|    0   N/A  N/A      2239      G   /usr/lib/xorg/Xorg                 60MiB |
|    0   N/A  N/A      2383      G   /usr/bin/gnome-shell              102MiB |
+-----------------------------------------------------------------------------+

And here is my "nvcc --version" output:

(rapids-21.10) ibayramoglu@ibayramoglu-MS-7C02:~$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243

**Comment by benfred** [comment]
How did you install nvtabular? It looks like the ```nvtabular.inference.triton.model_config_pb2``` file isn't being generated which should have been done by the install

**Comment by bayismet** [comment]
I'm not sure if I used pip or conda. What is the true way of a clean install? So I can follow that.

Thanks.

**Comment by rnyak** [comment]
@bayismet the recommended way is to use docker containers. 

First you need to pull the docker image: `docker pull nvcr.io/nvidia/merlin/merlin-pytorch-training:21.11` 

then follow the instructions [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples#running-the-example-notebooks) to launch the docker container and the jupyter notebooks.

Then please install public torch via `pip install torch==1.10.0`.

Hope that helps.

**Comment by rnyak** [comment]
@bayismet could you make the example notebooks work with docker images as suggested above? If yes, can we close this issue?

**Comment by bayismet** [comment]
Hello there!

TL;DR: Yes it solved my problem, so I'm closing the issue.

I can export the model and the workflow as ensembled, but rn, I'm having problems with inference, which is a totally different topic. So in the end, yes it solved my problem.

Thanks!
